File: Medium/2849. Determine if a Cell Is Reachable at a Given Time/2849. Determine if a Cell Is Reachable at a Given Time.js
/**
 * Determines if a cell is reachable at a given time.
 *
 * @param {number} sx - Starting x-coordinate
 * @param {number} sy - Starting y-coordinate
 * @param {number} fx - Ending x-coordinate
 * @param {number} fy - Ending y-coordinate
 * @param {number} t - Time in seconds
 * @return {boolean} - Returns true if the cell can be reached in exactly t seconds, or false otherwise
 */
var isReachableAtTime = function (sx, sy, fx, fy, t) {
    const dx = Math.abs(sx - fx);
    const dy = Math.abs(sy - fy);

    if (dx === 0 && dy === 0) {
        return t !== 1;
    } else {
        return Math.max(dx, dy) <= t;
    }
};

File: Medium/2849. Determine if a Cell Is Reachable at a Given Time/2849. Determine if a Cell Is Reachable at a Given Time.java
class Solution {
    public boolean isReachableAtTime(int sx, int sy, int fx, int fy, int t) {
        int dx = Math.abs(sx - fx);
        int dy = Math.abs(sy - fy);

        if (dx == 0 && dy == 0) {
            return t != 1;
        } else {
            return Math.max(dx, dy) <= t;
        }
    }
}
